agent/proxycfg/manager.go

```diff
@@ -40,49 +40,67 @@ type Manager struct {
 	stateCh chan struct{}
 
 	mu       sync.Mutex
-	proxies  map[string]*State
+	proxies  map[string]*state
 	watchers map[string]map[uint64]chan *ConfigSnapshot
 }
 
+// ManagerConfig holds the required external dependencies for a Manager
+// instance. All fields must be set to something valid or the manager will
+// panic.
```

Comment:
Would just be good to note since this is usually true: after initializing a manager with this config, the struct should not be read or modified further.

Reply:
I actually take the config by value and not pointer since its a tiny struct (4 pointers) anyway so the caller can't modify the value we store after construction. But mentioning that in docs is sensible!